File: app/src/main/java/com/riqsphere/myapplication/upcoming/UpcomingAdapter.java
package com.riqsphere.myapplication.upcoming;

import android.content.Context;
import android.media.Image;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.makeramen.roundedimageview.RoundedImageView;
import com.riqsphere.myapplication.R;

import java.util.ArrayList;

public class UpcomingAdapter extends RecyclerView.Adapter<UpcomingAdapter.ViewHolder> {

    private Context mContext;
    private ArrayList<UpcomingModel> mList;
    public UpcomingAdapter(Context context, ArrayList<UpcomingModel> list){
        mContext = context;
        mList = list;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {

        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.upcoming_card,parent,false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {

        (holder).BindView(position);
    }

    @Override
    public int getItemCount() {
        return mList.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder{
        RoundedImageView card_AnimeImage;
        TextView card_AnimeTittle;
        TextView card_EpTittle;
        TextView card_EpNum;
        TextView card_Time;


        public ViewHolder(View itemView){
            super(itemView);
            card_AnimeImage = itemView.findViewById(R.id.uc_image);
            card_AnimeTittle = itemView.findViewById(R.id.uc_anime_tittle);
            card_EpTittle = itemView.findViewById(R.id.uc_ep_tittle);
            card_EpNum = itemView.findViewById(R.id.uc_ep_num);
            card_Time = itemView.findViewById(R.id.uc_time);
        }

        public void BindView (int position){
            UpcomingModel ucCard = mList.get(position);

            card_AnimeImage.setImageResource(ucCard.getImage());
            card_AnimeTittle.setText(ucCard.getAnimeTittle());
            card_EpTittle.setText(ucCard.getEpTittle());
            card_EpNum.setText(ucCard.getEpNum());
            card_Time.setText(ucCard.getTime());
        }
    }
}

File: app/src/main/java/com/riqsphere/myapplication/upcoming/UpcomingAdapter.kt
package com.riqsphere.myapplication.upcoming

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.makeramen.roundedimageview.RoundedImageView
import com.riqsphere.myapplication.R
import com.riqsphere.myapplication.utils.ImageHandler
import java.util.*

class UpcomingAdapter(private val mContext: Context): RecyclerView.Adapter<UpcomingAdapter.ViewHolder>() {

    private var mList = ArrayList<UpcomingModel>()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view =
            LayoutInflater.from(parent.context).inflate(R.layout.upcoming_card, parent, false)
        return ViewHolder(view)
    }

    fun setData(upcoming: ArrayList<UpcomingModel>) {
        mList = upcoming
        notifyDataSetChanged()
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) = holder.bindView(position)

    override fun getItemCount() = mList.size

    inner class ViewHolder(itemView: View) :
        RecyclerView.ViewHolder(itemView) {
        private val cardAnimeImage: RoundedImageView = itemView.findViewById(R.id.uc_image)
        private val cardAnimeTitle: TextView = itemView.findViewById(R.id.uc_anime_title)
        private val cardEpTitle: TextView = itemView.findViewById(R.id.uc_ep_title)
        private val cardEpNum: TextView = itemView.findViewById(R.id.uc_ep_num)
        private val cardTime: TextView = itemView.findViewById(R.id.uc_time)

        fun bindView(position: Int) {
            val ucCard = mList[position]
            ImageHandler.getInstance(this@UpcomingAdapter.mContext).load(ucCard.imageURL).into(cardAnimeImage)
            cardAnimeTitle.text = ucCard.animeTitle
            cardEpNum.text = ucCard.epNum
            cardTime.text = ucCard.time
        }
    }

}